spec/latest/json-ld-framing/index.html

```diff
@@ -1089,7 +1089,10 @@ <h2>Frame Matching Algorithm</h2>
   of a set of properties, if neither <code>@id</code>, nor <code>@type</code> are in the frame.</p>
 
 <ol class="algorithm">
-  <li><em>Node</em> matches if it has an <code>@id</code> property including any <a>IRI</a> or <a>blank node</a> in the <code>@id</code> property in <em>frame</em>.
+  <li><em>Node</em> matches if it has an <code>@id</code> property
+    including any <a>IRI</a> or <a>blank node</a> in the <code>@id</code>
+    property in <em>frame</em>. Otherwise, if <em>frame</em> has a non-empty <code>@id</code> property
+    which is not an empty <a>dictionary</a>, <em>node</em> does not match.
```

Comment:
I had to read this a few times to get it, but I do now. It's essentially, do the first sentence to determine if the node definitely matches. Then, if it doesn't, you don't know yet if the node matches via some other rules. But you still need to check if it should be withheld entirely from matching by checking to see if the frame has an `@id` that doesn't match the node's `@id`. So there's some complexity in there that isn't explicit, but is still accurate.

I think we can leave it unless others also find it hard to parse in the future.

Reply:
Yes, really three options, match if, do  not match if, otherwise, fall through. Maybe changing to the following is easier to parse?

> Node matches if it has an `@id` property value which also a value of the `@id` property in frame. Otherwise, node does not match, if frame has a non-empty `@id` property, other than an empty dictionary. Otherwise, frame does not have an `@id` property, continue to the next step.